specs/public_inputs.md

```diff
@@ -0,0 +1,297 @@
+
+- EVM Circuit Block Table
+    - block[-1..-257].Hash
+- State Circuit
+    - block[-1].Root
+
+## Globals
+
+- ChainID: 64 bits
+
+### Circuits
+
+- EVM Circuit Block Table
+    - ChainID
+- TxCircuit
+    - ChainID
+
+## Transactions
+
+- Nonce: 64 bits
+- GasPrice: 256 bits
+- Gas: 64 bits
+- CallerAddress: 160 bits
+- CalleeAddress: 160 bits
+- IsCreate: 1 bit
+- Value: 256 bits
+- CallDataLength: 64 bits
+- CallData: CallDataLength bytes
+- TxSignHash: 256 bits
+- Signature
+    - v: 256 bits
+    - r: 256 bits
+    - s: 256 bits
+
+Fields used for RLP for TxSignHash:
+- Nonce
+- GasPrice
+- Gas
+- CalleeAddress
+- Value
+- CallData
+- ChainID
+
+Fields used for RLP for TxHash:
+- Nonce
+- GasPrice
+- Gas
+- CalleeAddress
+- Value
+- CallData
+- ChainID
+- Signature.v
+- Signature.r
+- Signature.s
+
+### Circuits
+
+- TxCircuit
+    - All Fields
+
+# Necessary public Data
+
+Some of the data defined in the previous section will be available along with
+the proof when it is submitted for verification.  More data than strictly
+necessary to verify a proof will be included in order to allow a node to
+synchronize the State Trie.
+
+## Data to synchronize the State Trie
+
+In order to synchronize the State Trie after a new block (assuming we have the state
+of the previous block), we need the at least the following data:
+
+- For each tx
+    - GasPrice: 256 bits
+    - Gas: 64 bits
+    - CallerAddress: 160 bits
+    - CalleeAddress: 160 bits
+    - Value: 256 bits
+    - CallData: CallDataLength bytes
+- Block fields that affect EVM execution
+    - Coinbase: 160 bits
+    - Difficulty: 256 bits
+    - Number: 64 bits
+    - GasLimit: 64 bits
+    - Time: 64 bits
+    - BaseFee: 256 bits
+- Extra fields that affect EVM execution
+    - block[-1..-257].Hash
+    - ChainID: 64 bits
+
+The signature is not needed to synchronize the State Trie, and the nonce can be
+inferred from the State Trie of the previous block.
+
+## Data to calculate a new block
+
+In order to calculate a new block (assuming we have the state of the previous
+block), we need at least the following data:
+
+- A way to prove that block[-1].Root (calculated from the known State Trie) is
+  included in block[-1].Hash
+    - A simple way to resolve this is by publishing the StateRoot with each
+      proof (liked via public input)
+    - Another way to resolve this is by publishing all block fields, so that a
+      verifier can calculate the block hash in the circuit proving that it uses
+      the expected StateRoot.
+
+## Data to verify synchronization
+
+Even though a node may have enough data to synchronize the State Trie, the node
+can't verify if the computed State Trie is correct (it's the same used in the
+proof).  For that it the following value should be published:
+
+- StateRoot
+
+# Public Input approach
+
+The necessary public data may not be equal to the verification circuit public
+inputs.  This is because every circuit public input value has a verification
```

Comment:
Ah, so I always use two spaces after a period :P  This is all over this documents and other documents I've contributed as well.  When rendering the markdown, this is not taken into account.

I would ask to leave this as is; and in case it's a concern that I use a different "spacing" style after periods, we can discuss it in an issue?